server/server.go

```diff
@@ -70,7 +70,7 @@ func New(opts *Opts) *Server {
 		IdleTimeout:         opts.IdleTimeout,
 		Dial:                opts.Dial,
 		Filter:              opts.Filter,
-		BufferSource:        buffers.Pool(),
+		BufferSource:        opts.BufferSource,
```

Comment:
This is the main change - Let the caller to specify the buffer source, or use the default. The pool for lampshade may not be optimal for other purposes.